src/query.ts

```diff
@@ -456,26 +438,36 @@ export class Query {
    * @param  {String}   queryName   The query_name parameter value to set
    *
    */
-  public queryName(queryName?: QueryTypes): void {
-    if (queryName) {
-      this.setParameter('query_name', queryName);
+  public queryName(queryName?: string): Query {
+    if (typeof queryName === 'string' || typeof queryName === 'undefined') {
+      if (queryName) {
+        this._queryName = queryName;
+      } else {
+        delete this._queryName;
+      }
     } else {
-      delete this.params.query_name;
+      throw new QueryValueError('Value error: queryname must be an string value');
     }
+    return this;
   }
 
   /**
    * Sets the nostats flag, call without parameters to clear it
    *
-   * @param  {Boolean}    nostats   Wether to send the nostats flag or not
+   * @param - nostats   Wether to send the nostats flag or not
    *
    */
-  public noStats(nostats?: boolean): void {
-    if (nostats && !!nostats) {
-      this.setParameter('nostats', 1);
+  public noStats(noStats?: boolean | number): Query {
```

Comment:
Don't allow numbers in the type definition if value must be a boolean. Just coerce to boolean unless the value is null or undefined. If the value must be a number in the query params, just coerce to boolean and use it to set the number.

```js
if (noStats == null) {
  this._noStats = undefined;
} else {
  this._noStats = !!noStats ? 1 : 0;
}
```

I wouldn't validate this more than that.